app/src/main/java/org/stepik/android/view/profile_certificates/ui/fragment/ProfileCertificatesFragment.kt

```diff
@@ -0,0 +1,132 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        injectComponent()
+
+        certificatesPresenter = ViewModelProviders
+            .of(this, viewModelFactory)
+            .get(ProfileCertificatesPresenter::class.java)
+
+        certificatesAdapter = DefaultDelegateAdapter()
+        certificatesAdapter += CertificateProfileAdapterDelegate(::onCertificateClicked)
+
+        Timber.d("User id: $userId")
+    }
+
+    private fun injectComponent() {
+        App.componentManager()
+            .profileComponent(userId)
+            .inject(this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? =
+        inflater.inflate(R.layout.fragment_profile_certificates, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        viewStateDelegate = ViewStateDelegate()
+        viewStateDelegate.addState<ProfileCertificatesView.State.Idle>()
+        viewStateDelegate.addState<ProfileCertificatesView.State.SilentLoading>()
+        viewStateDelegate.addState<ProfileCertificatesView.State.Loading>(view, profileCertificatesLoading)
+        viewStateDelegate.addState<ProfileCertificatesView.State.Error>(view, certificatesLoadingError)
+        viewStateDelegate.addState<ProfileCertificatesView.State.CertificatesCache>(view, profileCertificatesRecycler)
+        viewStateDelegate.addState<ProfileCertificatesView.State.CertificatesRemote>(view, profileCertificatesRecycler)
+        viewStateDelegate.addState<ProfileCertificatesView.State.NoCertificates>()
+
+        tryAgain.setOnClickListener { setDataToPresenter(forceUpdate = true) }
+        profileCertificatesTitle.setOnClickListener { screenManager.showCertificates(requireContext(), userId) }
+
+        profileCertificatesRecycler.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
+        profileCertificatesRecycler.isNestedScrollingEnabled = false
+        profileCertificatesRecycler.adapter = certificatesAdapter
+
+        setDataToPresenter()
+    }
+
+    private fun setDataToPresenter(forceUpdate: Boolean = false) {
+        certificatesPresenter.showCertificatesForUser(forceUpdate)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        certificatesPresenter.attachView(this)
+    }
+
+    override fun onStop() {
+        certificatesPresenter.detachView(this)
+        super.onStop()
+    }
+
+    private fun onCertificateClicked(path: String) {
+        screenManager.showPdfInBrowserByGoogleDocs(requireActivity(), path)
+    }
+
+    override fun setState(state: ProfileCertificatesView.State) {
+        viewStateDelegate.switchState(state)
+
+        when (state) {
+            is ProfileCertificatesView.State.Loading -> {
+                userId = state.userId
```

Comment:
In case of "my profile" tab on main screen:
 - first time component will be `profileComponent(0)` 
 - when screen recreated component will be `profileComponent(actualProfileId)` but parent component will still be `profileComponent(0)`, so `profileDataObservable` will not work